e2e/nomostest/config_sync.go

```diff
@@ -692,6 +695,9 @@ func RootSyncObjectV1Alpha1FromRootRepo(nt *NT, name string) *v1alpha1.RootSync
 	} else if rs.Spec.Override != nil {
 		rs.Spec.Override.ReconcileTimeout = nil
 	}
+	// Unregister the Helm repo for syncing, if any
+	rsKey := client.ObjectKeyFromObject(rs)
+	delete(nt.RootSyncHelmCharts, rsKey) // TODO: delete repo?
```

Comment:
Should we delete the helm chart from nt.RootSyncHelmCharts here?

If the helm chart is only added to `nt.RootSyncHelmCharts`, then we don't need the delete here.

In the case when a helm chart from `nt.RootSyncHelmCharts` is converted to a Git repo, we should delete it from `nt.RootSyncHelmCharts` with the conversion.